### PR TITLE
feat(alignment): add alignment layout

### LINKED ISSUE #251: Create layouts to align elements
Create a layout that align elements. For example on the card footer align btns to the right

### PR DESCRIPTION
**Pull request by matthewcarleton**
Alignment layout. This may need vertical alignment at some point as well, if so we will need to take a different approach to how this is built.

close #251

### CODE REVIEW

File: src/patternfly/layouts/Alignment/docs/code.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+## Overview
+
+Use the alignment layout when you want to align content to the left, right or center.
```
Comment (michael-coker) [inline src/patternfly/layouts/Alignment/docs/code.md]: oxford comma? :))))

File: src/patternfly/layouts/Alignment/examples/centerAlign.hbs [hunk 2]
```diff
@@ -0,0 +1 @@
+{{#> alignment pf-c-alignment--Modifier="pf-m-center" }} content {{/alignment}}
```
Comment (michael-coker) [inline src/patternfly/layouts/Alignment/examples/centerAlign.hbs]: I would make these blocks like this to better format them in the code block output.

```
{{#> alignment pf-c-alignment--Modifier="pf-m-center" }}
  content
{{/alignment}}
```

File: src/patternfly/layouts/Alignment/examples/index.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import Documentation from '@siteComponents/Documentation';
+import Example from '@siteComponents/Example';
+import LeftAlign from './leftAlign.hbs';
+import CenterAlign from './centerAlign.hbs';
+import RightAlign from './rightAlign.hbs';
```
[the remark this answers is not in the document]
Comment (michael-coker) [reply src/patternfly/layouts/Alignment/examples/index.js]: And the name after `import` should include the layout name (i.e., `import AlignmentLeftAlign`)

File: src/patternfly/layouts/Alignment/examples/index.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import docs from '../docs/code.md';
+import '../styles.scss';
+
+export const Docs = docs;
+
+export default () => {
+  const leftAlign = LeftAlign();
+  const centerAlign = CenterAlign();
+  const rightAlign = RightAlign();
```
Comment (michael-coker) [inline src/patternfly/layouts/Alignment/examples/index.js]: The var name should include the layout name (i.e., `const alignmentLeftAlign`)

File: src/patternfly/layouts/Alignment/styles.scss [hunk 5]
```diff
@@ -0,0 +1,17 @@
+@import "../../patternfly-utilities";
+.pf-l-alignment {
+  padding: 0;
+  margin: 0;
+
+  &.pf-m-left {
+    text-align: left;
+  }
```
Comment (michael-coker) [inline src/patternfly/layouts/Alignment/styles.scss]: add a space after this line

File: src/patternfly/layouts/Alignment/styles.scss [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+  &.pf-m-center {
+    text-align: center;
+  }
+
+  &.pf-m-right {
+    text-align: right;
+  }
+
```
Comment (michael-coker) [inline src/patternfly/layouts/Alignment/styles.scss]: remove this blank line

File: src/patternfly/layouts/Alignment/styles.scss [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+}
```
Comment (michael-coker) [inline src/patternfly/layouts/Alignment/styles.scss]: add a blank line at the end

File: src/patternfly/layouts/Alignment/examples/centerAlign.hbs (hunk 2 above)
Comment (matthewcarleton) [reply src/patternfly/layouts/Alignment/examples/centerAlign.hbs]: I had it like that but the formatter doesn't like it :(

Comment (michael-coker) [reply src/patternfly/layouts/Alignment/examples/centerAlign.hbs]: Which formatter? Looks like the bullseye and gallery examples are formatted this way, but the rest of the layouts (and components) aren't.

**Comment by srambach** [comment]
Looks good to me. I agree we'll probably have to address vertical alignment sooner than later.

**Pull request merged**